sequencer/finalizer.go

```diff
@@ -81,52 +108,105 @@ func (f *finalizer) Start() {
 		for {
 			tx := f.worker.GetBestFittingTx(wipBatch.remainingResources)
 			if tx != nil {
+				lenOfTxs := len(wipBatch.txs)
+				isFirstTx := lenOfTxs == 1
 
-				// If it is not the first tx in the batch GER should be 0x00
-				ger := wipBatch.GER
-
-				// TODO: Populate processBatchRequest dinamic fields
+				var ger common.Hash
+				if isFirstTx {
+					ger = wipBatch.GER
+				} else {
+					ger = state.ZeroHash
+				}
 
+				// TODO: Populate processBatchRequest dynamic fields
+				processBatchRequest.GlobalExitRoot = ger
+				processBatchRequest.IsFirstTx = isFirstTx
 				result := f.executor.ExecuteTransaction(processBatchRequest)
+
 				// executionResult := execute tx (only passing to the executor this tx, starting at currentBatch.intermediaryRoot)
-				// if ko:
-				// // decide if we [MoveTxToNotReady, DeleteTx, UpdateTx]
-				// if ok:
-				// // currentBatch.remainingResources.sub(executionResult.Counters)
-				// // if resourced overflows:
-				// // // f.worker.UpdateTx()
-				// // // continue (TODO: actually no, because we need to check the timeouts)
-				// // add tx to current batch
-				// // send tx to the DB (already finalized) through channel (async store)
-				// // remove tx from pool (already finalized) through channel (async store)
-				// // f.worker.UpdateAfterSingleSuccessfulTxExecution()
+				if result.Error != nil {
+					// // decide if we [MoveTxToNotReady, DeleteTx, UpdateTx]
+
+				} else {
+					processedTx := result.Responses[0]
+					from, err := types.Sender(types.NewEIP155Signer(processedTx.Tx.ChainId()), &processedTx.Tx)
+					if err != nil {
+						from, err = types.Sender(types.HomesteadSigner{}, &processedTx.Tx)
+					}
+
+					err = wipBatch.remainingResources.Sub(RemainingResources{
+						remainingZKCounters: result.UsedZkCounters,
+						remainingBytes:      uint64(processedTx.Tx.Size()),
+						remainingGas:        processedTx.GasUsed,
+					})
+					if err != nil {
+						f.worker.UpdateTx(processedTx.TxHash, from, wipBatch.remainingResources.remainingZKCounters)
+						// (TODO: check the timeouts)
+						continue
+					}
+
+					// We have a successful processing if we are here
+					wipBatch.intermediaryStateRoot = result.NewStateRoot
+					processBatchRequest.StateRoot = result.NewStateRoot
+					processBatchRequest.OldAccInputHash = result.NewAccInputHash
+					// We store the processed transaction, add it to the batch and delete from the pool atomically.
+					go f.dbManager.StoreProcessedTxAndDeleteFromPool(ctx, wipBatch.batchNumber, processedTx)
+					go f.worker.UpdateAfterSingleSuccessfulTxExecution(from, result.TouchedAddresses)
```

Comment:
```suggestion
					f.worker.UpdateAfterSingleSuccessfulTxExecution(from, result.TouchedAddresses)
```

We decided to do this synchronously. The idea is to compromise a bit of performance but to gain more security (improve "hit ratio")

Reply:
Yep, sorry mistake here.